garden_ai/client.py

```diff
@@ -295,9 +299,29 @@ def create_pipeline(
         return Pipeline(**data)
 
     def register_model(self, local_model: LocalModel) -> RegisteredModel:
-        registered_model = upload_to_model_registry(local_model)
-        local_data.put_local_model(registered_model)
-        return registered_model
+        if "S3_UPLOAD_FEATURE_FLAG" in os.environ:
```

Comment:
This is where the upload feature is "hidden" for now. If you don't turn on this env variable, behavior should be exactly the same after this PR.